src/DynamoCore/Engine/Profiling/IProfilingData.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using Dynamo.Graph.Nodes;
+
+namespace Dynamo.Engine.Profiling
+{
+    /// <summary>
+    /// Returns information about time spent compiling and exectuing nodes.                                                 
+    /// </summary>
+    public interface IProfilingData
```

Comment:
hmm, maybe we make this more specific or marked as `X` for experimental, my thought here is that we might want to add more information to this interface and if we release with it public we'll have to worry about API changes. Thoughts? Maybe we call it `IProfilingExecutionTimeData` so we can wrap it up as a member of a larger interface later?